trigger/task.go

```diff
@@ -664,51 +645,41 @@ func (t *taskTriggers) taskRegressionByTest(sub *event.Subscription) (*notificat
 	if t.task.Requester != evergreen.RepotrackerVersionRequester || !isFailedTaskStatus(t.task.Status) {
 		return nil, nil
 	}
+	// if no tests, alert only if it's a regression in task status
+	if len(t.task.LocalTestResults) == 0 {
+		return t.taskRegression(sub)
+	}
 
+	catcher := grip.NewBasicCatcher()
 	previousCompleteTask, err := task.FindOne(task.ByBeforeRevisionWithStatuses(t.task.RevisionOrderNumber,
 		task.CompletedStatuses, t.task.BuildVariant, t.task.DisplayName, t.task.Project))
 	if err != nil {
 		return nil, errors.Wrap(err, "error fetching previous task")
 	}
 
-	record, err := alertrecord.FindByLastTaskRegressionByTestWithNoTests(sub.ID, t.task.DisplayName, t.task.BuildVariant, t.task.Project, t.task.RevisionOrderNumber)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to fetch alertrecord")
+	if previousCompleteTask != nil {
+		t.oldTestResults = mapTestResultsByTestFile(previousCompleteTask)
 	}
 
-	catcher := grip.NewBasicCatcher()
-	// if no tests, alert only if it's a regression in task status
-	if len(t.task.LocalTestResults) == 0 {
-		if record != nil && !isTaskStatusRegression(record.TaskStatus, t.task.Status) {
-			return nil, nil
+	testsToAlert := []task.TestResult{}
+	for i := range t.task.LocalTestResults {
+		if !testMatchesRegex(t.task.LocalTestResults[i].TestFile, sub) {
+			continue
 		}
-		catcher.Add(alertrecord.InsertNewTaskRegressionByTestWithNoTestsRecord(sub.ID, t.task.Id, t.task.DisplayName, t.task.Status, t.task.BuildVariant, t.task.Project, t.task.RevisionOrderNumber))
-
-	} else {
-		if previousCompleteTask != nil {
-			t.oldTestResults = mapTestResultsByTestFile(previousCompleteTask)
-		}
-
-		testsToAlert := []task.TestResult{}
-		for i := range t.task.LocalTestResults {
-			if !testMatchesRegex(t.task.LocalTestResults[i].TestFile, sub) {
-				continue
-			}
-			var shouldInclude bool
-			shouldInclude, err = t.shouldIncludeTest(sub.ID, previousCompleteTask, &t.task.LocalTestResults[i])
-			if err != nil {
-				catcher.Add(err)
-				continue
-			}
-			if shouldInclude {
-				testsToAlert = append(testsToAlert, t.task.LocalTestResults[i])
-			}
+		var shouldInclude bool
+		shouldInclude, err = t.shouldIncludeTest(sub.ID, previousCompleteTask, &t.task.LocalTestResults[i])
+		if err != nil {
+			catcher.Add(err)
+			continue
 		}
-		if len(testsToAlert) == 0 {
-			return nil, nil
+		if shouldInclude {
+			testsToAlert = append(testsToAlert, t.task.LocalTestResults[i])
 		}
-		// TODO EVG-3416 use testsToAlert in message formatting
 	}
+	if len(testsToAlert) == 0 {
+		return nil, nil
+	}
+	// TODO EVG-3416 use testsToAlert in message formatting
```

Comment:
what is this?

Reply:
It's a todo that Richard added. Since that ticket is closed but doesn't address this, I opened EVG-5317 and erased the todo